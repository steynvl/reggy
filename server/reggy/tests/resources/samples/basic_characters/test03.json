{
  "sampleStringsInfo": [
    [
      {
        "markerType": "Basic characters",
        "markedStrings": [
          "asas"
        ],
        "markerInfo": {
          "caseInsensitive": true,
          "lowerCaseLetters": true,
          "upperCaseLetters": true,
          "digits": true,
          "punctuationAndSymbols": true,
          "matchAllExceptSpecified": false,
          "whiteSpace": true,
          "lineBreaks": true,
          "individualCharacters": "steyn"
        },
        "repeatInfo": {
          "repeat": "5"
        }
      }
    ]
  ],
  "generalRegexInfo": {
    "startRegexMatchAt": "Anywhere",
    "endRegexMatchAt": "Anywhere",
    "regexTarget": "Java"
  },
  "generateMethod": "samplesAndSemantics"
}
