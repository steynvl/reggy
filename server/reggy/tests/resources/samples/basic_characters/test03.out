Java
Pattern regex = Pattern.compile("[A-Za-z\\d\\p{Punct}\\s\\r\\nsteyn]{5}", Pattern.CASE_INSENSITIVE);
(?i)[A-Za-z\\d\\p{Punct}\\s\\r\\nsteyn]{5}
Perl
my $regex = /[A-Za-z\d\p{PosixPunct}\s\r\nsteyn]{5}/i;
(?i)[A-Za-z\d\p{PosixPunct}\s\r\nsteyn]{5}
POSIX
(?i)[A-Za-z\d\p{P}\s\r\nsteyn]{5}
(?i)[A-Za-z\d\p{P}\s\r\nsteyn]{5}
Python
regex = re.compile(r'[A-Za-z\d\p{P}\s\r\nsteyn]{5}', re.IGNORECASE)
(?i)[A-Za-z\d\p{P}\s\r\nsteyn]{5}
JavaScript
const regex = /[A-Za-z\d[!"\#$%&'()*+,\-./:;<=>?@\[\\]^_`{|}~]\s\r\nsteyn]{5}/i;
(?i)[A-Za-z\d[!"\#$%&'()*+,\-./:;<=>?@\[\\]^_`{|}~]\s\r\nsteyn]{5}
PHP
$regex = '/[A-Za-z\d\p{P}\s\r\nsteyn]{5}/i';
(?i)[A-Za-z\d\p{P}\s\r\nsteyn]{5}
Golang
regex, _ := regexp.Compile("(?i)[A-Za-z\\d\\p{P}\\s\\r\\nsteyn]{5}")
(?i)[A-Za-z\\d\\p{P}\\s\\r\\nsteyn]{5}
Rust
let re = Regex::new(r"(?i)[A-Za-z\d\p{P}\s\r\nsteyn]{5}").unwrap();
(?i)[A-Za-z\d\p{P}\s\r\nsteyn]{5}
C#
var re = new Regex(@"[A-Za-z\d\p{P}\s\r\nsteyn]{5}", RegexOptions.IgnoreCase);
(?i)[A-Za-z\d\p{P}\s\r\nsteyn]{5}
Scala
val re = "(?i)[A-Za-z\\d\\p{Punct}\\s\\r\\nsteyn]{5}".r
(?i)[A-Za-z\\d\\p{Punct}\\s\\r\\nsteyn]{5}
Kotlin
val regex = Regex("[A-Za-z\\d\\p{Punct}\\s\\r\\nsteyn]{5}", IGNORE_CASE)
(?i)[A-Za-z\\d\\p{Punct}\\s\\r\\nsteyn]{5}